crates/services/p2p/src/peer_report.rs

```diff
@@ -118,340 +96,107 @@ impl PeerReportBehaviour {
                 }
             }
-            FromSwarm::DialFailure(e) => {
-                let (ping_handler, identity_handler) = e.handler.into_inner();
-                let ping_event = DialFailure {
-                    peer_id: e.peer_id,
-                    handler: ping_handler,
-                    error: e.error,
-                };
-                let identity_event = DialFailure {
-                    peer_id: e.peer_id,
-                    handler: identity_handler,
-                    error: e.error,
-                };
-                self.heartbeat
-                    .on_swarm_event(FromSwarm::DialFailure(ping_event));
-                self.identify
-                    .on_swarm_event(FromSwarm::DialFailure(identity_event));
-            }
-            FromSwarm::ListenFailure(e) => {
-                let (ping_handler, identity_handler) = e.handler.into_inner();
-                let ping_event = ListenFailure {
-                    handler: ping_handler,
-                    local_addr: e.local_addr,
-                    send_back_addr: e.send_back_addr,
-                };
-                let identity_event = ListenFailure {
-                    handler: identity_handler,
-                    local_addr: e.local_addr,
-                    send_back_addr: e.send_back_addr,
-                };
-                self.heartbeat
-                    .on_swarm_event(FromSwarm::ListenFailure(ping_event));
-                self.identify
-                    .on_swarm_event(FromSwarm::ListenFailure(identity_event));
-            }
-            _ => {
-                self.heartbeat.handle_swarm_event(&event);
-                self.identify.handle_swarm_event(&event);
-            }
+            FromSwarm::DialFailure(e) => {}
+            FromSwarm::ListenFailure(e) => {}
+            _ => {}
         }
     }
 
+    fn on_connection_handler_event(
+        &mut self,
+        _peer_id: PeerId,
+        _connection_id: ConnectionId,
+        _event: THandlerOutEvent<Self>,
+    ) {
+    }
+
     fn poll(
         &mut self,
         cx: &mut Context<'_>,
-        params: &mut impl PollParameters,
-    ) -> Poll<NetworkBehaviourAction<Self::OutEvent, Self::ConnectionHandler>> {
+    ) -> Poll<ToSwarm<Self::ToSwarm, THandlerInEvent<Self>>> {
         if let Some(event) = self.pending_events.pop_front() {
-            return Poll::Ready(NetworkBehaviourAction::GenerateEvent(event))
-        }
-
-        match self.heartbeat.poll(cx, params) {
-            Poll::Pending => {}
-            Poll::Ready(action) => {
-                let action =
-                    <PeerReportBehaviour as FromAction<Heartbeat>>::convert_action(
-                        self, action,
-                    );
-                if let Some(action) = action {
-                    return Poll::Ready(action)
-                }
-            }
-        }
-
-        loop {
-            // poll until we've either exhausted the events or found one of interest
```

Comment:
This is something that doesn't have parity with the changes. I just let the generated `poll` (from the `NetworkBehavior` macro) do its thing. That might be a problem.

We could manually implement `poll` for `FuelBehavior` to do the same thing as this, but that might mean we have to implement _all_ the methods, and some of them are hairy with the nested `Selects` etc.